### PR TITLE
[MM-44475] Team Unarchive: Do not allow to unarchive if they have reached the limit of teams

### PR DESCRIPTION
**Pull request by nickmisasi**
#### Summary
If a user attempts to "restore" (ie, "unarchive") a team, we must check if they are at or over the limit, and disallow them to do so. 

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-44475

#### Release Note

```release-note
User's will be prevented from unarchiving teams if they're at or over the teams limit for their plan.
```

### CODE REVIEW

File: api4/team_test.go [hunk 1]
```diff
@@ -578,6 +581,54 @@ func TestRestoreTeam(t *testing.T) {
...
+				Active: model.NewInt(1),
+			},
+		}, nil).Once()
+		team := createTeam(t, true, model.TeamOpen)
+		_, resp, err := client.RestoreTeam(team.Id)
+		require.Error(t, err)
+		CheckBadRequestStatus(t, resp)
+	})
+
+	t.Run("cloud below limit returns 200", func(t *testing.T) {
```
Comment (marianunez) [inline api4/team_test.go:609]: Can we add tests to cover the cases of not cloud and of empty limits ?

**Review by marianunez — APPROVED** [review]
LGTM, non-blocking comment below

File: api4/team_test.go (hunk 1 above)
Comment (nickmisasi) [reply api4/team_test.go:609]: Already tests in the suite for the base cases

File: model/client4.go [hunk 2]
```diff
@@ -8110,3 +8110,16 @@ func (c *Client4) GetPostsUsage() (*PostsUsage, *Response, error) {
 	err = json.NewDecoder(r.Body).Decode(&usage)
 	return usage, BuildResponse(r), err
 }
+
+// GetTeamsUsage returns rounded off total usage of teams for the instance
```
Comment (vish9812) [inline model/client4.go:8114]:
```suggestion
// GetTeamsUsage returns total usage of teams for the instance
```

**Review by vish9812 — APPROVED** [review]

**Review by stevemudie — APPROVED** [review]
Deferring for testing post merge to allow full on review alongside https://github.com/mattermost/mattermost-server/pull/20282

**Comment by nickmisasi** [comment]
@marianunez @neallred made a few small changes to this PR so just looking for another once over

File: api4/team_test.go [hunk 3]
```diff
@@ -67,27 +70,103 @@ func TestCreateTeam(t *testing.T) {
...
+				Active: model.NewInt(200),
+			},
+		}, nil).Once()
+		team := &model.Team{Name: GenerateTestUsername(), DisplayName: "Some Team", Type: model.TeamOpen}
+		_, resp, err := th.Client.CreateTeam(team)
+		require.NoError(t, err)
+		CheckCreatedStatus(t, resp)
+	})
+
+	t.Run("non below limit returns 200", func(t *testing.T) {
```
Comment (neallred) [inline api4/team_test.go:146]: I might be missing something, but are the `cloud below limit returns 200` and `non below limit returns 200` tests identical? Maybe there was some other case that was intended to be tested, after copy-paste the new test wasn't written? Or maybe it can be deleted?

**Review by neallred — APPROVED** [review]

**Review by marianunez — APPROVED** [review]
LGTM

File: api4/team_test.go (hunk 3 above)
Comment (nickmisasi) [reply api4/team_test.go]: Yeah it's identical, removed!

**Comment by marianunez** [comment]
@nickmisasi once done with the set of PRs for this feature, let's cherry pick to `cloud` branch
